tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2336,44 +2379,28 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
         false
     }
 
-    val hijackedClassMethodToHelperName: Map[String, String] = Map(
-        "getClass__jl_Class" -> "objectGetClass",
-        "toString__T"        -> "objectToString",
-        "clone__O"           -> "objectClone",
-        "finalize__V"        -> "objectFinalize",
-        "notify__V"          -> "objectNotify",
-        "notifyAll__V"       -> "objectNotifyAll",
-        "equals__O__Z"       -> "objectEquals",
-        "hashCode__I"        -> "objectHashCode",
-
-        "length__I"                          -> "charSequenceLength",
-        "charAt__I__C"                       -> "charSequenceCharAt",
-        "subSequence__I__I__jl_CharSequence" -> "charSequenceSubSequence",
-
-        "compareTo__O__I"            -> "comparableCompareTo",
-        "compareTo__jl_Boolean__I"   -> "comparableCompareTo",
-        "compareTo__jl_Character__I" -> "comparableCompareTo",
-        "compareTo__jl_Byte__I"      -> "comparableCompareTo",
-        "compareTo__jl_Short__I"     -> "comparableCompareTo",
-        "compareTo__jl_Integer__I"   -> "comparableCompareTo",
-        "compareTo__jl_Long__I"      -> "comparableCompareTo",
-        "compareTo__jl_Float__I"     -> "comparableCompareTo",
-        "compareTo__jl_Double__I"    -> "comparableCompareTo",
-        "compareTo__jl_String__I"    -> "comparableCompareTo",
-
-        "booleanValue__Z" -> "booleanBooleanValue",
-
-        "charValue__C" -> "characterCharValue",
-
-        "byteValue__B"   -> "numberByteValue",
-        "shortValue__S"  -> "numberShortValue",
-        "intValue__I"    -> "numberIntValue",
-        "longValue__J"   -> "numberLongValue",
-        "floatValue__F"  -> "numberFloatValue",
-        "doubleValue__D" -> "numberDoubleValue",
-
-        "isNaN__Z"      -> "isNaN",
-        "isInfinite__Z" -> "isInfinite"
+    def typeToBoxedHijackedClass(tpe: Type): String = (tpe: @unchecked) match {
+      case ClassType(cls) => cls
+      case AnyType        => Definitions.ObjectClass
+      case UndefType      => Definitions.BoxedUnitClass
+      case BooleanType    => Definitions.BoxedBooleanClass
+      case CharType       => Definitions.BoxedCharacterClass
+      case ByteType       => Definitions.BoxedByteClass
+      case ShortType      => Definitions.BoxedShortClass
+      case IntType        => Definitions.BoxedIntegerClass
+      case LongType       => Definitions.BoxedLongClass
+      case FloatType      => Definitions.BoxedFloatClass
+      case DoubleType     => Definitions.BoxedDoubleClass
+      case StringType     => Definitions.StringClass
+    }
+
+    /* Ideally, we should figure out this set using a CHA. However, this is
```

Comment:
CHA?

Reply:
Class Hierarchy Analysis. I thought that was a sufficiently well-known term. I'll elaborate.